src/dmd/dtoh.d

```diff
@@ -0,0 +1,1724 @@
+        if (vd.type.ty == AST.Tsarray)
+            return false;
+        return true;
+    }
+
+    override void visit(AST.StructDeclaration sd)
+    {
+        debug (Debug_DtoH)
+        {
+            printf("[AST.StructDeclaration enter] %s\n", sd.toChars());
+            scope(exit) printf("[AST.StructDeclaration exit] %s\n", sd.toChars());
+        }
+        if (sd.isInstantiated())
+            return;
+        if (cast(void*)sd in visited)
+            return;
+        if (!sd.type || !sd.type.deco)
+            return;
+        //version(BUILD_COMPILER)
+        //{
+            if (sd.getModule() && sd.getModule().isIgnoredModule())
+                return;
+        //}
+
+        visited[cast(void*)sd] = true;
+        if (linkage != LINK.c && linkage != LINK.cpp)
+        {
+            buf.printf("// ignoring non-cpp struct %s because of linkage\n", sd.toChars());
+            return;
+        }
+
+        buf.writestring(sd.isUnionDeclaration() ? "union" : "struct");
+        pushAlignToBuffer(sd.alignment);
+        buf.writestring(sd.ident.toChars());
+        if (sd.members)
+        {
+            buf.writestring("\n{\n");
+            auto save = adparent;
+            adparent = sd;
+            foreach (m; *sd.members)
+            {
+                m.accept(this);
+            }
+            adparent = save;
+            // Generate default ctor
+            if (!sd.noDefaultCtor)
+            {
+                buf.printf("    %s()", sd.ident.toChars());
+                size_t varCount;
+                bool first = true;
+                foreach (m; *sd.members)
+                {
+                    if (auto vd = m.isVarDeclaration())
+                    {
+                        if (!memberField(vd))
+                            continue;
+                        varCount++;
+
+                        if (!vd._init && !vd.type.isTypeBasic() && !vd.type.isTypePointer && !vd.type.isTypeStruct &&
+                            !vd.type.isTypeClass && !vd.type.isTypeDArray && !vd.type.isTypeSArray)
+                        {
+                            continue;
+                        }
+                        if (vd._init && vd._init.isVoidInitializer())
+                            continue;
+
+                        if (first)
+                        {
+                            buf.printf(" : ");
+                            first = false;
+                        }
+                        else
+                        {
+                            buf.printf(", ");
+                        }
+                        buf.printf("%s(", vd.ident.toChars());
+
+                        if (vd._init)
+                        {
+                            AST.initializerToExpression(vd._init).accept(this);
+                        }
+                        buf.printf(")");
+                    }
+                }
+                buf.printf(" {}\n");
+            }
+
+            version (none)
```

Comment:
any ideas what this version none ins for?

Reply:
Not sure but the versioned code generates an additional constructor to set all fields, e.g.

```
S2(_d_int a, _d_int b, _d_long c) { this->a = a; this->b = b; this->c = c; }
```

for 

```
extern (C++) struct S2
{
    int a = 42;
    int b;
    long c;

    this(int a) {}
}
```